integration-tests/utils/integration-utils.js

```diff
@@ -190,9 +190,27 @@ const dismissNotifications = async page => {
 }
 
 const signIntoTerra = async (page, { token, testUrl }) => {
-  !!testUrl && await page.goto(testUrl, waitUntilLoadedOrTimeout(60 * 1000))
-  await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout: 60 * 1000 })
-  await waitForNoSpinners(page)
+  if (!!testUrl) {
+    const timeout = 30 * 1000
+    const pageLoadedPromise = Promise.race([
+      page.waitForXPath('//*[@id="signInButton"]', { visible: true, timeout }),
+      page.waitForXPath('//a[@href="#workspaces"]', { visible: true, timeout })
+    ])
+    const loadPagePromise = Promise.all([
+      page.goto(testUrl, waitUntilLoadedOrTimeout(timeout)),
+      pageLoadedPromise
+    ])
+      .then(() => waitForNoSpinners(page))
+
+    try {
+      await loadPagePromise
+    } catch (e) {
+      console.log(`Warning: Page loading timed out during sign in. Reloading URL: "${testUrl}"`)
```

Comment:
Is it worth logging `e` in case the cause of failure is something else?

Reply:
Sure, I'll add that. 
I was thinking of same thing. But I didn't do it because if reload has worked, I don't really care to see it. If still fails on the second attempt, (same) error will be thrown.